### PR TITLE
Node widget using new CustomWidget API

### PR DESCRIPTION
**Pull request by stjahns**
Notes: 
- Node will probably need to be converted to part of an Enum if we want more than one custom widget type
- Having trouble having the custom widget play nicely with the builtin conrod widgets, not sure if need to setup/initialize things differently
- Not actually using custom color, frame color, and frame size when building the element yet

### CODE REVIEW

File: src/node.rs [hunk 1]
```diff
@@ -75,7 +82,7 @@ impl<'a> CustomWidget for Node<'a> {
 
     type State = NodeState;
 
-    fn set<C>(self, ui_id: UiId, ui: &mut Ui<C>) {
+    fn update<C>(self, state: NodeState, ui_id: UiId, ui: &mut Ui<C, Node<'a>>) -> WidgetUpdate<NodeState> {
```
Comment (mitchmindtree) [inline src/node.rs:85]: You might have to add `where C: CharacterCache` here

CharacterCache is not yet exposed (it's in `graphics::character::CharacterCache`, but we should expose it from the conrod lib too). I'll do it now.

edit: This should help with the weird errors you were getting in main.rs

Comment (stjahns) [inline src/node.rs:85]: ahh that makes sense!

**Comment by viperscape** [comment]
still can't compile, unless my cargo is acting up:
`node.rs:89:13: 89:31 error: unresolved import `conrod::WidgetKind`. There is no `WidgetKind` in `conrod`
node.rs:89         use conrod::WidgetKind;
                       ^~~~~~~~~~~~~~~~~~
error: aborting due to previous error
Could not compile `mush`.`
[widgetkind](https://github.com/stjahns/mush/blob/node-widget/src/node.rs#L89)

**Comment by stjahns** [comment]
@viperscape - i think you just need to pull the last commit, I removed the reference to WidgetKind

**Comment by stjahns** [comment]
@viperscape - get on IRC :)

**Comment by stjahns** [comment]
never mind, i didn't push that yet.. sorry

**Comment by stjahns** [comment]
Merged here - https://github.com/viperscape/mush/pull/5/files

**Comment by viperscape** [comment]
@stjahns I wasn't able to get on irc, shoulda been tho bc it would have cleared up my confusion :)

**Pull request merged**